src/game/shell.c

```diff
@@ -199,21 +296,26 @@ bool Shell_MakeScreenshot()
         break;
     }
 
-    char path[20];
-    for (int i = 0; i < 10000; i++) {
-        sprintf(path, "screenshot%04d.%s", i, ext);
-        char *full_path = NULL;
-        File_GetFullPath(path, &full_path);
+    // Screenshot name
+    char ss_name[LEVEL_TITLE_SIZE + 25]; // 50
+    Shell_GetScreenshotName(ss_name, ext);
 
-        if (!File_Exists(full_path)) {
-            bool result = Output_MakeScreenshot(full_path);
-            Memory_Free(full_path);
-            full_path = NULL;
-            return result;
-        }
+    // Screenshot folder
+    char ss_folder[SS_FOLDER_NAME_SIZE];
+    sprintf(ss_folder, "screenshots");
+    File_CreateDirectory(ss_folder);
+
+    // Screenshot folder/name path
+    char path[LEVEL_TITLE_SIZE + SS_FOLDER_NAME_SIZE + 63]; // 100
+    sprintf(path, "%s/%s", ss_folder, ss_name);
 
+    char *full_path = NULL;
+    File_GetFullPath(path, &full_path);
+    if (!File_Exists(full_path)) {
```

Comment:
Please flip the condition so that we exit early in case of errors to reduce indentation.